cmake/CPackConfig.cmake

```diff
@@ -79,4 +79,15 @@ install(FILES "${PRECICE_PACKAGING_DIR}/lintian-override"
   RENAME ${CPACK_PACKAGE_NAME}
   )
 
+# Configure a pkg-config file for the debian package
+configure_file(
+  "${PROJECT_SOURCE_DIR}/tools/packaging/debian/precice.pc.in"
+  "${PRECICE_PACKAGING_DIR}/pkgconfig/${CPACK_PACKAGE_NAME}.pc"
```

Comment:
We have an issue here: The result file is stored (for me) as `libprecice1.3.0.pc`.

This means that I also need to use it as:

```
# This works
PKG_CONFIG_ALLOW_SYSTEM_CFLAGS=1 pkg-config --cflags libprecice1.3.0

# This does not work
PKG_CONFIG_ALLOW_SYSTEM_CFLAGS=1 pkg-config --cflags libprecice
```

The issue for us is that we would need to update the version in every adapter all the time (assuming we don't use a proper build system). Looking at my list of installed packages, some omit the version. Why not do the same for us?

```
$ ls /usr/lib/x86_64-linux-gnu/pkgconfig/
atk.pc                 egl.pc                  gio-2.0.pc                harfbuzz.pc       libdrm_nouveau.pc   libssl.pc      pangoxft.pc                 Qt5Gui.pc                QtGui.pc          spice-client-glib-2.0.pc  xcb-present.pc  xkbcommon.pc
cairo-fc.pc            expat.pc                gio-unix-2.0.pc           hdf5-openmpi.pc   libdrm.pc           libsystemd.pc  pixman-1.pc                 Qt5Network.pc            QtHelp.pc         spice-controller.pc       xcb-randr.pc    xkbcomp.pc
cairo-ft.pc            fftw3f.pc               glesv2.pc                 hdf5.pc           libdrm_radeon.pc    libudev.pc     protobuf-lite.pc            Qt5OpenGLExtensions.pc   QtNetwork.pc      tic.pc                    xcb-render.pc   xmu.pc
cairo-gobject.pc       fftw3l.pc               glib-2.0.pc               hdf5-serial.pc    libffi.pc           libxml-2.0.pc  protobuf.pc                 Qt5OpenGL.pc             QtOpenGL.pc       tinfo.pc                  xcb-shape.pc    xorg-wacom.pc
cairo.pc               fftw3.pc                gl.pc                     hwloc.pc          libjpeg.pc          menu.pc        pthread-stubs.pc            Qt5PrintSupport.pc       QtScript.pc       wayland-client.pc         xcb-shm.pc      xrandr.pc
cairo-pdf.pc           fftw3q.pc               glu.pc                    ice.pc            libpcre16.pc        mirclient.pc   python-2.7.pc               Qt5Sql.pc                QtScriptTools.pc  wayland-cursor.pc         xcb-sync.pc     xrender.pc
cairo-png.pc           fontconfig.pc           gmodule-2.0.pc            icu-i18n.pc       libpcre32.pc        mircookie.pc   python2.pc                  Qt5Test.pc               QtSql.pc          wayland-egl.pc            xcb-xfixes.pc   xscrnsaver.pc
cairo-ps.pc            form.pc                 gmodule-export-2.0.pc     icu-io.pc         libpcrecpp.pc       mircore.pc     python-3.5m.pc              Qt5Widgets.pc            QtSvg.pc          wayland-scanner.pc        xcomposite.pc   xshmfence.pc
cairo-script.pc        freetype2.pc            gmodule-no-export-2.0.pc  icu-le.pc         libpcre.pc          ncurses.pc     python-3.5.pc               Qt5XcbQpa.pc             QtTest.pc         wayland-server.pc         xcursor.pc      xt.pc
cairo-svg.pc           fuse.pc                 gobject-2.0.pc            icu-lx.pc         libpcreposix.pc     ncurses++.pc   python3.pc                  Qt5Xml.pc                QtUiTools.pc      x11.pc                    xdamage.pc      xxf86vm.pc
cairo-tee.pc           gdk-2.0.pc              gthread-2.0.pc            icu-uc.pc         libpng12.pc         netcdf.pc      python.pc                   QtCLucene.pc             QtWebKit.pc       x11-xcb.pc                xdmcp.pc        zlib.pc
cairo-xcb.pc           gdk-pixbuf-2.0.pc       gtk+-2.0.pc               intel-gen4asm.pc  libpng.pc           openssl.pc     Qt3Support.pc               QtCore.pc                QtXmlPatterns.pc  xau.pc                    xext.pc
cairo-xcb-shm.pc       gdk-pixbuf-xlib-2.0.pc  gtk+-unix-print-2.0.pc    libcrypto.pc      libprecice1.3.0.pc  panel.pc       Qt5Concurrent.pc            QtDBus.pc                QtXml.pc          xcb-dri2.pc               xfixes.pc
cairo-xlib.pc          gdk-x11-2.0.pc          gtk+-x11-2.0.pc           libcurl.pc        librlog.pc          pangocairo.pc  Qt5Core.pc                  QtDeclarative.pc         ruby-2.3.pc       xcb-dri3.pc               xft.pc
cairo-xlib-xrender.pc  geany.pc                harfbuzz-gobject.pc       libdrm_amdgpu.pc  libselinux.pc       pangoft2.pc    Qt5DBus.pc                  QtDesignerComponents.pc  ruby.pc           xcb-glx.pc                xinerama.pc
dri.pc                 geoclue-2.0.pc          harfbuzz-icu.pc           libdrm_intel.pc   libsepol.pc         pango.pc       Qt5EglDeviceIntegration.pc  QtDesigner.pc            sm.pc             xcb.pc                    xi.pc
```

Reply:
Clearly, we need a version number in the pkgconfig, but maybe we can assume that we only keep the latest installed one. So, the name could be generic, but the version could change every time.